src/coreclr/jit/importer.cpp

```diff
@@ -9696,6 +9696,7 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                     }
                     if (helperNode != nullptr)
                     {
+                        helperNode->gtFlags |= GTF_ORDER_SIDEEFF;
```

Comment:
Why do we need to set the ordering effect on these?

The calls should not be reorderable with `GTF_GLOB_REF`-tagged field static access by themselves.